### PR TITLE
Fix deepseek r1 use graph cause accuracy issue

### PR DESCRIPTION
**Pull request by zichen-kuuga**
1. When enabling graph, the types of gating_output and hidden_states may cause accuracy anomalies. Here, they are converted to float32 in advance
2. Integrating the moe_fused_gate of mate, enabling eligible MoE models to utilize the fused kernel

### CODE REVIEW

File: vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        hidden_states = hidden_states.to(dtype=torch.float32)
+    if gating_output.dtype != torch.float32:
+        gating_output = gating_output.to(dtype=torch.float32)
+    # ========================== END ==========================
+    if (
+        gating_output.shape[1] // num_expert_group <= 32
+        or (
+            num_expert_group == 1 and gating_output.shape[1] in {160, 256, 384}
+        )  # XXX (MUSA): will support more cases in the future
+    ) and is_power_of_two(e_score_correction_bias.shape[0]):
```
[the remark this answers is not in the document]
Comment (zichen-kuuga) [reply vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py]: The judgment that e_score_correction_bias is not none has been added

File: vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
+import math
+
+import torch
+from mate import moe_fused_gate as mate_moe_fused_gate
```
[the remark this answers is not in the document]
Comment (zichen-kuuga) [reply vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py]: Already compatible with the judgment that mate is missing

File: vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        and (
+            gating_output.shape[1] // num_expert_group <= 32
+            or (
+                num_expert_group == 1 and gating_output.shape[1] in {160, 256, 384}
+            )  # XXX (MUSA): will support more cases in the future
+        )
+        and (
+            e_score_correction_bias is not None
+            and is_power_of_two(e_score_correction_bias.shape[0])
+        )
```
Comment (yeahdongcn) [inline vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py]: Why are they excluded from `MUSA ADAPTATION`?

File: vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            routed_scaling_factor if routed_scaling_factor is not None else 1.0,
+            renormalize,
+            apply_routed_scaling_factor_on_output,
+        )
+
+        return topk_weights, topk_ids
+
+    assert hidden_states.size(0) == gating_output.size(0), "Number of tokens mismatch"
+
+    if scoring_func == "softmax":
```
Comment (yeahdongcn) [inline vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py:138]: I couldn't find them in the vllm `0.17.0` tag.

Comment (zichen-kuuga) [reply vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py:138]: The implementation of grouped_topk in 0.17.0 is here: https://github.com/vllm-project/vllm/blob/v0.17.0/vllm/model_executor/layers/fused_moe/router/grouped_topk_router.py#L114-L165

File: vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py (hunk 3 above)
Comment (zichen-kuuga) [reply vllm_musa/model_executor/layers/fused_moe/router/grouped_topk_router.py]: This part should be placed in the MUSA ADAPTATION, and has been updated

**Review by yeahdongcn — APPROVED** [review]

**Comment by yeahdongcn** [comment]
Cherry-picked to `v0.18.0-dev` as well.
